ml-agents/mlagents/trainers/trainer/rl_trainer.py

```diff
@@ -140,11 +174,12 @@ def _checkpoint(self) -> NNCheckpoint:
             logger.warning(
                 "Trainer has multiple policies, but default behavior only saves the first."
             )
-        policy = list(self.policies.values())[0]
-        model_path = policy.model_path
-        settings = SerializationSettings(model_path, self.brain_name)
-        checkpoint_path = os.path.join(model_path, f"{self.brain_name}-{self.step}")
-        policy.checkpoint(checkpoint_path, settings)
+        # policy = list(self.policies.values())[0]
+        # model_path = policy.model_path
+        settings = SerializationSettings(self.saver.model_path, self.brain_name)
+        checkpoint_path = os.path.join(self.saver.model_path, f"{self.brain_name}-{self.step}")
+        # policy.checkpoint(checkpoint_path, settings)
+        self.saver.save_checkpoint(checkpoint_path, settings)
```

Comment:
This code is rather strange. Would it be simpler to have save_checkpoint take as input the brain_name and the steps (and calculate the path from it)

Reply:
Was doing that and then see the line below NNCheckpoint is also using this path. Would it be better if returning this path from save_checkpoint() and feed it to NNCheckpoint?

Reply:
I think that would be better, the logic into `checkpoint_path = os.path.join(self.saver.model_path, f"{self.brain_name}-{self.step}")` should be part of the saver.

Reply:
fixed that.